src/api/base.ts

```diff
@@ -1,39 +1,35 @@
-const baseUrl = `http://localhost:8000/api/`;
+import { apiUrl } from "./util";
+import { refresh } from "./token";
 
-const apiUrl = (endpoint: string) => {
-    return `${baseUrl}${endpoint}`;
-};
-
-export const fetchToken = async (data: { username: string, password: string }) => {
-    const response = await fetch(apiUrl(`token/`), {
-        method: `POST`,
-        headers: {
-            "Content-Type": `application/json`
-        },
-        body: JSON.stringify(data)
-    });
-    const token = await response.json();
-
-    if (token.access) {
-        localStorage.setItem(`token`, token.access);
+export const fetchFunds = async () => {
+    const loggedIn = await refresh();
+    if (!loggedIn.success) {
+        return {
+            success: false,
+            message: loggedIn.message
+        };
     }
-    return token;
-};
 
-export const fetchFunds = async () => {
     const response = await fetch(apiUrl(`funds/`), {
         method: `GET`,
         headers: {
             "Content-Type": `application/json`,
-            "Authorization": `Bearer ${localStorage.getItem(`token`)}`,
+            "Authorization": `Bearer ${localStorage.getItem(`access_token`)}`,
         }
     });
 
     const json = await response.json();
 
+    console.log(json);
+
     if (json.detail) {
-        // TODO: https://github.com/ksiondag/base_webapp/issues/3
-        localStorage.removeItem(`token`);
+        return {
+            success: false,
+            message: json.detail
+        };
     }
-    return json;
+    return {
+        success: true,
+        funds: json
```

Comment:
Working on more consistent structure of api input/output that is decoupled from how the actual backend responds (so that if the backend changes it all stays clean).